<h1>ACTIVITY 1: Palmer Penguins Data Analytics Project<h1>
<h3>Analyst: Janelle Kaye Borabo<h3>
<hr>


#libraries

In [1]:
import numpy as np
import pandas as pd
import warnings 

In [2]:
warnings.filterwarnings('ignore')

In [3]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [4]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1>Descriptive Analytics</h1>

In [5]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

In [6]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h1>Filtering a Cloumn/Data</h1>

In [7]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [8]:
penguins['species'] == "Adelie"

0       True
1       True
2       True
3       True
4       True
       ...  
339    False
340    False
341    False
342    False
343    False
Name: species, Length: 344, dtype: bool

In [9]:
penguins[penguins['species'] == "Adelie"]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [10]:
Adelie_df=penguins[penguins['species'] == "Adelie"]

In [11]:
Adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [12]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h1>Gentoo</h1>

In [13]:
Gentoo_df=penguins[penguins['species'] == "Gentoo"]

In [14]:
Gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

<h1>Chinstrap</h1>

In [15]:
Chinstrap_df=penguins[penguins['species'] == "Chinstrap"]

In [16]:
Chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

<h1>Adelie Body Mass</h1>

In [17]:
Adelie_df['body_mass_g'].mean()

np.float64(3700.662251655629)

<h1>Adding Two data</h1>

In [18]:
penguins[(penguins['body_mass_g'] >= 4000) & (penguins['species'] == 'Adelie')]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
7    Adelie  Torgersen            39.2           19.6              195.0   
9    Adelie  Torgersen            42.0           20.2              190.0   
14   Adelie  Torgersen            34.6           21.1              198.0   
17   Adelie  Torgersen            42.5           20.7              197.0   
19   Adelie  Torgersen            46.0           21.5              194.0   
35   Adelie      Dream            39.2           21.1              196.0   
39   Adelie      Dream            39.8           19.1              184.0   
43   Adelie      Dream            44.1           19.7              196.0   
45   Adelie      Dream            39.6           18.8              190.0   
49   Adelie      Dream            42.3           21.2              191.0   
51   Adelie     Biscoe            40.1           18.9              188.0   
53   Adelie     Biscoe            42.0           19.5              200.0   
61   Adelie     Biscoe            41.3           21.1              195.0   
63   Adelie     Biscoe            41.1           18.2              192.0   
67   Adelie     Biscoe            41.1           19.1              188.0   
69   Adelie  Torgersen            41.8           19.4              198.0   
73   Adelie  Torgersen            45.8           18.9              197.0   
75   Adelie  Torgersen            42.8           18.5              195.0   
79   Adelie  Torgersen            42.1           19.1              195.0   
81   Adelie  Torgersen            42.9           17.6              196.0   
83   Adelie  Torgersen            35.1           19.4              193.0   
91   Adelie      Dream            41.1           18.1              205.0   
93   Adelie      Dream            39.6           18.1              186.0   
95   Adelie      Dream            40.8           18.9              208.0   
97   Adelie      Dream            40.3           18.5              196.0   
99   Adelie      Dream            43.2           18.5              192.0   
101  Adelie     Biscoe            41.0           20.0              203.0   
103  Adelie     Biscoe            37.8           20.0              190.0   
109  Adelie     Biscoe            43.2           19.0              197.0   
111  Adelie     Biscoe            45.6           20.3              191.0   
113  Adelie     Biscoe            42.2           19.5              197.0   
115  Adelie     Biscoe            42.7           18.3              196.0   
125  Adelie  Torgersen            40.6           19.0              199.0   
127  Adelie  Torgersen            41.5           18.3              195.0   
129  Adelie  Torgersen            44.1           18.0              210.0   
133  Adelie      Dream            37.5           18.5              199.0   
139  Adelie      Dream            39.7           17.9              193.0   
146  Adelie      Dream            39.2           18.6              190.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g   sex  year  
7         4675.0  male  2007  
9         4250.0   NaN  2007  
14        4400.0  male  2007  
17        4500.0  male  2007  
19        4200.0  male  2007  
35        4150.0  male  2007  
39        4650.0  male  2007  
43        4400.0  male  2007  
45        4600.0  male  2007  
49        4150.0  male  2007  
51        4300.0  male  2008  
53        4050.0  male  2008  
61        4400.0  male  2008  
63        4050.0  male  2008  
67        4100.0  male  2008  
69        4450.0  male  2008  
73        4150.0  male  2008  
75        4250.0  male  2008  
79        4000.0  male  2008  
81        4700.0  male  2008  
83        4200.0  male  2008  
91        4300.0  male  2008  
93        4450.0  male  2008  
95        4300.0  male  2008  
97        4350.0  male  2008  
99        4100.0  male  2008  
101       4725.0  male  2009  
103       4250.0  male  2009  
109       4775.0  male  2009  
111       4600.0  male  2009  
11

In [19]:
penguins[(penguins['flipper_length_mm'] >= 200) & (penguins['species'] == 'Adelie')]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
53   Adelie     Biscoe            42.0           19.5              200.0   
90   Adelie      Dream            35.7           18.0              202.0   
91   Adelie      Dream            41.1           18.1              205.0   
95   Adelie      Dream            40.8           18.9              208.0   
101  Adelie     Biscoe            41.0           20.0              203.0   
123  Adelie  Torgersen            41.4           18.5              202.0   
129  Adelie  Torgersen            44.1           18.0              210.0   
137  Adelie      Dream            40.2           20.1              200.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
53        4050.0    male  2008  
90        3550.0  female  2008  
91        4300.0    male  2008  
95        4300.0    male  2008  
101       4725.0    male  2009  
123       3875.0    male  2009  
129       4000.0    male  2009  
137       3975.0    male  2009  
151       4000.0    male  2009

<h1>How to Fill Up Null Values</h1>

In [20]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [21]:
penguins['bill_length_mm']=penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [22]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [23]:
penguins['bill_depth_mm']=penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [24]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [25]:
penguins['flipper_length_mm']=penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [26]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [27]:
penguins['body_mass_g']=penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [28]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [29]:
penguins['sex']=penguins['sex'].fillna(penguins['sex'].mode()[0])

In [30]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h1>Data Types</h1>

In [31]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [32]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      344.000000     344.000000         344.000000   344.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.443643       1.969027          14.020657   799.613058   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.275000      15.600000         190.000000  3550.000000   
50%         44.250000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

<h1>Changing Data types</h1>

In [33]:
penguins['species']=penguins['species'].astype('category')
penguins['year']=penguins['year'].astype('category')
penguins['island']=penguins['island'].astype('category')
penguins['sex']=penguins['sex'].astype('category')

In [34]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


In [35]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      344.000000     344.000000         344.000000   344.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.443643       1.969027          14.020657   799.613058
min         32.100000      13.100000         172.000000  2700.000000
25%         39.275000      15.600000         190.000000  3550.000000
50%         44.250000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

<h2>1. What is the body mass of the heaviest Adelie Penguins in Dream Island ?</h2>
<hr>


In [36]:
adelie_df = penguins[(penguins['species'] == "Adelie") & (penguins['island'] == 'Dream')]

In [37]:
adelie_df['body_mass_g'].mean()

np.float64(3688.3928571428573)

<h2>Insight no.1</h2>
<h3> In Dream Island, the Heaviest Adelie Penguin has the body mass of 3688g</h3>

<h2>2. Which Island has the most Gentoo Penguins?</h2>
<hr>

In [38]:
island_df = penguins[penguins['species'] == "Gentoo"]

In [39]:
island_df.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      124.000000     124.000000         124.000000   124.000000
mean        47.475983      14.999606         217.055768  5068.965761
std          3.086123       0.996447           6.621801   508.164382
min         40.900000      13.100000         200.915205  3950.000000
25%         45.275000      14.200000         212.000000  4700.000000
50%         47.250000      15.000000         216.000000  5000.000000
75%         49.525000      15.725000         221.000000  5500.000000
max         59.600000      17.300000         231.000000  6300.000000

In [40]:
island_df.value_counts()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  sex     year
Gentoo   Biscoe  40.9            13.7           214.0              4650.0       female  2007    1
                 41.7            14.7           210.0              4700.0       female  2009    1
                 42.0            13.5           210.0              4150.0       female  2007    1
                 42.6            13.7           213.0              4950.0       female  2008    1
                 42.7            13.7           208.0              3950.0       female  2008    1
                                                                                               ..
                 53.4            15.8           219.0              5500.0       male    2009    1
                 54.3            15.7           231.0              5650.0       male    2008    1
                 55.1            16.0           230.0              5850.0       male    2009    1
                 55.9    

In [41]:
islands_df = penguins[(penguins['species'] == "Gentoo") & (penguins['island'] == 'Biscoe')].value_counts()

In [42]:
islands_df.value_counts()

count
1    124
Name: count, dtype: int64

<h2>Insight no.2</h2>
<h3> Biscoe Island was the only Island has Gentoo Penguin with 124 penguins</h3>

<h2>3. What is the body mass of the heaviest Gentoo Penguins in Biscoe Island ?</h2>
<hr>

In [43]:
biscoe_df = penguins[(penguins['species'] == "Gentoo") & (penguins['island'] == 'Biscoe')]

In [44]:
biscoe_df['body_mass_g'].mean()

np.float64(5068.965761177136)

<h2>Insight no.3</h2>
<h3> In Biscoe Island, the Heaviest Gentoo Penguin has the body mass of 5068g</h3>

<h2>4. What is the body mass of the heaviest Chinstrap Penguins in Dream Island ?</h2>
<hr>

In [45]:
torgersen_df = penguins[(penguins['species'] == "Chinstrap") & (penguins['island'] == 'Dream')]

In [46]:
torgersen_df['body_mass_g'].mean()

np.float64(3733.0882352941176)

In [47]:
penguins['island']

0      Torgersen
1      Torgersen
2      Torgersen
3      Torgersen
4      Torgersen
         ...    
339        Dream
340        Dream
341        Dream
342        Dream
343        Dream
Name: island, Length: 344, dtype: category
Categories (3, object): ['Biscoe', 'Dream', 'Torgersen']

<h2>Insight no.4</h2>
<h3> In Dream Island, the Heaviest Chinstrap Penguin has the body mass  of 3733g</h3>

<h2>5. Which sex has the longer bill length in Chinstrap Penguins?</h2>
<hr>

In [48]:
f_bill_df = penguins[(penguins['species'] == "Chinstrap") & (penguins['sex'] == 'female') ]

In [49]:
f_bill_df['bill_length_mm'].mean()

np.float64(46.57352941176471)

In [50]:
m_bill_df = penguins[(penguins['species'] == "Chinstrap") & (penguins['sex'] == 'male') ]

In [51]:
m_bill_df['bill_length_mm'].mean()

np.float64(51.09411764705882)

<h2>Insight no.5</h2>
<h3> In Chinstrap Penguin, the male has the longer Bill Length with the average of 51mm</h3>

<h2>6. Which year has the most Adelie Penguins?</h2>
<hr>

In [52]:
species_df = penguins[penguins['species'] == "Adelie" ]

In [53]:
species_df['year'].value_counts()

year
2009    52
2007    50
2008    50
Name: count, dtype: int64

<h2>Insight no.6</h2>
<h3> Year 2009 has the most Adelie penguins.</h3>

<h2>7. What is the lighest Gentoo Penguins in Biscoe Island?</h2>
<hr>

In [54]:
gentoo_df = penguins[(penguins['species'] == "Gentoo") & (penguins['island'] == 'Biscoe')]

In [55]:
gentoo_df['body_mass_g'].min()

np.float64(3950.0)

<h2>Insight no.7</h2>
<h3> In Biscoe Island, the lighest Gentoo Penguin have 3950g </h3>

<h2>8. Is there correlation between bill depth to body mass?</h2>
<hr>

In [62]:
flip_cor = penguins['bill_depth_mm'].corr(penguins['body_mass_g'])

In [64]:
flip_cor

np.float64(-0.4719156211860664)

<h2>Insight no.8</h2>
<h3>There is a no correlation between penguins bill depth and thier body mass.</h3>

<h2>9. Is there correlation between bill length to flipper length?</h2>
<hr>

In [68]:
bill_cor = penguins['bill_length_mm'].corr(penguins['flipper_length_mm'])

In [69]:
bill_cor

np.float64(0.6561813407464278)

<h2>Insight no.9</h2>
<h3>There is a no correlation between penguins bill depth and thier flipper length.</h3>

<h2>10. Is there correlation between flipper length to year?</h2>
<hr>

In [72]:
flip_cor = penguins['flipper_length_mm'].corr(penguins['year'])

In [73]:
flip_cor

np.float64(0.16893420971651854)

<h2>Insight no.10</h2>
<h3>There is a no correlation between penguins flipper length to year.</h3>

<h2>11. Which year has the most heaviest Adelie Penguins?</h2>
<hr>

In [79]:
year_adelie_df = penguins[(penguins['species'] == "Adelie") & (penguins['year'] == 2007)]

In [80]:
year_adelie_df['body_mass_g'].mean()

np.float64(3706.535087719298)

In [82]:
year1_adelie_df = penguins[(penguins['species'] == "Adelie") & (penguins['year'] == 2008)]

In [83]:
year1_adelie_df['body_mass_g'].mean()

np.float64(3742.0)

In [87]:
year2_adelie_df = penguins[(penguins['species'] == "Adelie") & (penguins['year'] == 2009)]

In [88]:
year2_adelie_df['body_mass_g'].mean()

np.float64(3664.903846153846)

<h2>Insight no.11</h2>
<h3> The year 2008 has the most heaviest Adelie Penguins.</h3>

<h2>12. Which year has the most lighest Gentoo Penguins?</h2>
<hr>

In [89]:
year_gentoo_df = penguins[(penguins['species'] == "Gentoo") & (penguins['year'] == 2007)]

In [94]:
year_gentoo_df['body_mass_g'].min()

np.float64(4100.0)

In [95]:
year1_gentoo_df = penguins[(penguins['species'] == "Gentoo") & (penguins['year'] == 2008)]

In [96]:
year1_gentoo_df['body_mass_g'].min()

np.float64(3950.0)

In [98]:
year2_gentoo_df = penguins[(penguins['species'] == "Gentoo") & (penguins['year'] == 2009)]

In [100]:
year2_gentoo_df['body_mass_g'].min()

np.float64(4201.754385964912)

<h2>Insight no.12</h2>
<h3> The year 2008 has the most lighest Gentoo Penguins.</h3>

<h2> Data Aggregation</h2>
<hr>

In [56]:
sp_is_x = penguins.groupby(['species','island','year']).size()

In [57]:
sp_is_x

species    island     year
Adelie     Biscoe     2007    10
                      2008    18
                      2009    16
           Dream      2007    20
                      2008    16
                      2009    20
           Torgersen  2007    20
                      2008    16
                      2009    16
Chinstrap  Biscoe     2007     0
                      2008     0
                      2009     0
           Dream      2007    26
                      2008    18
                      2009    24
           Torgersen  2007     0
                      2008     0
                      2009     0
Gentoo     Biscoe     2007    34
                      2008    46
                      2009    44
           Dream      2007     0
                      2008     0
                      2009     0
           Torgersen  2007     0
                      2008     0
                      2009     0
dtype: int64

In [58]:
avg_body = penguins.groupby(['species','island','year'])['body_mass_g'].mean()

In [59]:
avg_body

species    island     year
Adelie     Biscoe     2007    3620.000000
                      2008    3627.777778
                      2009    3857.812500
           Dream      2007    3671.250000
                      2008    3756.250000
                      2009    3651.250000
           Torgersen  2007    3785.087719
                      2008    3856.250000
                      2009    3489.062500
Chinstrap  Biscoe     2007            NaN
                      2008            NaN
                      2009            NaN
           Dream      2007    3694.230769
                      2008    3800.000000
                      2009    3725.000000
           Torgersen  2007            NaN
                      2008            NaN
                      2009            NaN
Gentoo     Biscoe     2007    5070.588235
                      2008    5019.565217
                      2009    5119.358054
           Dream      2007            NaN
                      2008            NaN
       

<h2>Determinig Correlation Using Pandas Pearson R</h2>

In [60]:
mass_flip_cor = penguins['flipper_length_mm'].corr(penguins['body_mass_g'])

In [61]:
mass_flip_cor

np.float64(0.8712017673060114)